finagle/h2/src/main/scala/com/twitter/finagle/buoyant/h2/BufferedStream.scala

```diff
@@ -89,6 +89,9 @@ class BufferedStream(underlying: Stream, bufferCapacity: Long = 16383) { buffere
             }
           }
         }
+        stream.onCancel.onSuccess { rst =>
+          underlying.cancel(rst)
+        }
```

Comment:
I'm not clear enough on the semantics of `fork()` to know that this is safe...  This means that if any reader cancels then all other readers will also be reset? How does this interact with retries?  Does this need to refcount underlying?

Reply:
The semantics of resets when retries are involved aren't really externally specified so it's somewhat up to us.  I think it makes sense to reset everything once a single reader cancels.

Consider when the client which sends a small but unterminated stream.  If the server resets that stream, I would interpret that to be a signal to the client to stop producing.

Reply:
On the other hand, consider that resets can be used to signal things like rate limiting.

I think this behavior is Okay for now, but this may warrant some more thought in general.